acorn/src/expression.js

```diff
@@ -431,13 +421,40 @@ pp.parseExprAtom = function(refDestructuringErrors) {
   }
 }
 
-pp.parseDynamicImport = function() {
+pp.parseExprImport = function(noDynamicImport) {
   const node = this.startNode()
-  this.next()
-  if (this.type !== tt.parenL) {
+  this.next() // skip `import`
+  switch (this.type) {
+  case tt.parenL:
+    if (noDynamicImport) {
+      this.raise(this.start, "Cannot use new with import()")
+    }
+    return this.parseDynamicImport(node)
+  default:
     this.unexpected()
   }
-  return this.finishNode(node, "Import")
+}
+
+pp.parseDynamicImport = function(node) {
+  this.next() // skip `(`
+
+  // Parse node.source.
+  if (this.type === tt.ellipsis) {
```

Comment:
Why are the explicit checks (the ellipsis and also the comma below) for things that would raise an error anyway? I'd prefer the terse variant that just allows the callee methods to raise the errors.

Reply:
This was the same behavior as before this PR. I will remove it.